hazelcast/src/main/java/com/hazelcast/instance/GroupProperties.java

```diff
@@ -221,6 +227,37 @@
      */
     public static final String PROP_JCACHE_PROVIDER_TYPE = "hazelcast.jcache.provider.type";
 
+    /**
+     * Result size limit for query operations on maps.
+     * <p/>
+     * This value defines the maximum number of returned elements for a single query result. If a query exceeds this number of
+     * elements a {@link QueryResultSizeExceededException} will be thrown.
+     * <p/>
+     * This feature is in place to prevent an OOME if a single node is requesting the whole data set of the cluster, e.g. by
```

Comment:
The local pre-check doesn't work if the query won't return all values (it's just summing up the partition sizes). So we need that check for `TruePredicate` there. The check in `QueryOperation` checks the real query results, so it will also work for all types of queries.